aten/src/ATen/core/op_registration/adaption.h

```diff
@@ -59,5 +62,41 @@ check_tensor_options_and_extract_memory_format(
     return options.memory_format_opt();
   }
 }
+
+TORCH_API void common_device_check_failure(optional<Device>& common_device, const at::Tensor& tensor, at::CheckedFrom methodName, at::CheckedFrom argName);
+
+inline void check_and_update_common_device(optional<Device>& common_device, const at::Tensor& tensor, at::CheckedFrom methodName, at::CheckedFrom argName) {
+  // TODO: Remove this check once all tensors are guaranteed to be defined
```

Comment:
demonstrated by https://github.com/pytorch/pytorch/pull/56564. Should I open an issue to track this? @ezyang

Reply:
Yes plz.